src/Atmos/Model/orientation.jl

```diff
@@ -18,6 +18,13 @@ gravitational_potential(::Orientation, aux::Vars) = aux.orientation.Φ
 ∇gravitational_potential(::Orientation, aux::Vars) = aux.orientation.∇Φ
 altitude(orientation::Orientation, aux::Vars) = gravitational_potential(orientation, aux) / grav
 vertical_unit_vector(orientation::Orientation, aux::Vars) = ∇gravitational_potential(orientation, aux) / grav
+function projection_normal(orientation::Orientation, aux::Vars, u⃗::AbstractVector) 
+  n̂ = vertical_unit_vector(orientation, aux)
+  return n̂ * (n̂' * u⃗)
+end
+function projection_tangential(orientation::Orientation, aux::Vars, u⃗::AbstractVector) 
```

Comment:
These functions are good to have. But I wonder whether we should make them more general. We'll need these (or functions like these) at boundaries in a number of contexts. Then the projections should be normal and tangential to the boundary, i.e., with `n̂` being the inward pointing normal vector on the boundary. (This notation for `n̂` is also consistent with the design docs. There and elsewhere, we used `k` for the vertical unit vector.)

Reply:
Sriharsha is working on the projection functions today.